ci/codecov.sh

```diff
@@ -75,23 +75,23 @@ elif [[ "$coverage_action" == "collect" ]] || [[ "$coverage_action" == "upload"
     : "${LCOV_BRANCH_COVERAGE:=1}" # Set default
 
     # coverage files are in ../../b2 from this location
-    lcov ${LCOV_OPTIONS} --rc lcov_branch_coverage="${LCOV_BRANCH_COVERAGE}" --gcov-tool="$GCOV" --directory "$BOOST_ROOT" --capture --output-file all.info
+    lcov ${LCOV_OPTIONS} --rc branch_coverage="${LCOV_BRANCH_COVERAGE}" --gcov-tool="$GCOV" --directory "$BOOST_ROOT" --capture --output-file all.info
```

Comment:
It looks like `--rc branch_coverage` should go to `LCOV_OPTIONS`

In the original code, still above, we have a switch to check for version 2+ so there we can use the old or new option to be sure we don't break anyone.